### ISSUE TITLE
x/build/cmd/coordinator: gitmirror health check is flaky because it reports status of a single gitmirror instance, as determined by load balancer

### ISSUE THREAD

**Issue by dmitshur** [golang/go#37828]
https://farmer.golang.org/status/gitmirror is intermittently reporting an issue with git mirroring of the main repository:

```
# "gitmirror" status: Git mirroring
# Notes: https://github.com/golang/build/tree/master/cmd/gitmirror
Error: repo go: hung? no activity since last success 4h13m30.458762593s ago
```

[![image](https://user-images.githubusercontent.com/1924134/76560381-6bc24100-6477-11ea-822a-f9af19d74fe7.png)](https://farmer.golang.org/)

It's normal for it to not have activity for some number of minutes, but 4+ hours seems unexpected.

**Edit:** This may be a false positive of some sort. I refreshed the status of https://farmer.golang.org/status/gitmirror just a few minutes after starting to type this, and it suddenly became "ok". I refreshed again, and it came back to reporting a problem. Now it's back to "ok". It should not be going from "ok" to "no activity for 4+ hours".

/cc @cagedmantis @toothrot

**Comment by toothrot** [comment]
There are 2 gitmirror instances. This was resolved by restarting the failing gitmirror instance (deleting the pod), which we discovered by proxying to the pods and investigating their statuses.

One idea would be to add stackdriver metrics to gitmirror for monitoring this. Another would be to improve the health check so that it checks all the pods, rather than relying on the kubedns load balancing to land on a healthy one.

**Comment by dmitshur** [comment]
This issue is happening again now. Refreshing farmer.golang.org sometimes shows that Git mirroring is okay, other times it's showing "repo go: hung? no activity since last success 8h41m49.610989692s ago":

![image](https://user-images.githubusercontent.com/1924134/77969839-b7b71780-72b8-11ea-8fba-3a712e44cd2c.png)

There isn't a real mirroring problem, because a 4-hour commit in the main Go repo is successfully mirrored to GitHub.

From looking at the logs for the 2 gitmirror instances, I see a number of errors from the Gerrit servers today. That may be what caused at least one of the instances to get into a bad state.

**Comment by dmitshur** [comment]
> to improve the health check so that it checks all the pods

I've sent [CL 226678](https://golang.org/cl/226678) that implements this.

**Comment by dmitshur** [comment]
Tested the CL above, it worked as expected:

![image](https://user-images.githubusercontent.com/1924134/78098327-c1ba4280-73ac-11ea-9585-a467fb1663a8.png)

No more flakiness, refreshing the page always produces the same result. 🎉

I've redeployed gitmirror to resolve its current issue. Now that the reporting at https://farmer.golang.org/#health is reliable, we can deal with gitmirror problems if they come up in the future.
